### PR TITLE
Add a `forc-submit` plugin command for submitting txs to a given node

### LINKED ISSUE #3875: Create a `forc submit` command for submitting serialized transactions to a fuel network
As a follow-up to #3582, and as mentioned in #3491, while `forc tx` handles transaction construction, `forc submit` (open to better names) would handle transaction submission. Following this, we'll be ready to refactor the `forc run` and `forc deploy` commands.

Usage would be something like:

```console
forc submit --url https://node-beta-2.fuel.network tx.json
```

The command should print the resulting `TransactionStatus` logs to stdout.

### PR DESCRIPTION
**Pull request by mitchmindtree**
Closes #3875.

### CODE REVIEW

**Review by JoshuaBatty — DISMISSED** [review]

File: forc-plugins/forc-submit/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        serde_json::from_reader(reader)?
+    } else {
+        let tx_bytes = std::fs::read(path)?;
+        let (_bytes, tx) = fuel_tx::Transaction::try_from_bytes(&tx_bytes)?;
+        tx
+    };
+    Ok(tx)
+}
+
+/// Print the transaction status in a more
```
Comment (bingcicle) [inline forc-plugins/forc-submit/src/lib.rs:61]: incomplete comment?

File: forc-plugins/forc-submit/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    /// Output the resulting transaction status as JSON rather than the default output.
+    #[clap(long = "tx-status-json", default_value_t = false)]
+    pub json: bool,
+}
+
+/// Deserialize a `Transaction` from the given file into memory.
+pub fn read_tx(path: &std::path::Path) -> anyhow::Result<fuel_tx::Transaction> {
+    let file = std::fs::File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let tx: fuel_tx::Transaction = if path.ends_with("json") {
```
Comment (bingcicle) [inline forc-plugins/forc-submit/src/lib.rs:51]: do we want to maybe handle an `Err` here for unsupported formats other than json/bin?

**Review by bingcicle — CHANGES_REQUESTED** [review]

File: forc-plugins/forc-submit/src/lib.rs (hunk 2 above)
Comment (mitchmindtree) [reply forc-plugins/forc-submit/src/lib.rs:51]: Good call, thanks!

**Comment by mitchmindtree** [comment]
Converting back to draft as I'm realising this command is likely better included as a part of the `forc-client` crate. Will address this soon!

**Comment by mitchmindtree** [comment]
OK I've rebased and integrated `forc submit` into the `forc-client` crate as a binary target (like `run` and `deploy`). Should be good to go now!

File: forc-plugins/forc-client/src/op/submit.rs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        println!("{id}");
+    }
+    Ok(())
+}
+
+/// Deserialize a `Transaction` from the given file into memory.
+pub fn read_tx(path: &std::path::Path) -> anyhow::Result<fuel_tx::Transaction> {
+    let file = std::fs::File::open(path)?;
+    let reader = std::io::BufReader::new(file);
+    let tx: fuel_tx::Transaction = if path.ends_with("json") {
```
Comment (kayagokalp) [inline forc-plugins/forc-client/src/op/submit.rs:33]: While testing,  I realized that even if the file has `json` or `bin` extension it fails with invalid file format error so, maybe we should be using `.extension()` for checking the extension ([relevant std reference](https://doc.rust-lang.org/std/path/struct.Path.html#method.ends_with)).

**Review by kayagokalp — CHANGES_REQUESTED** [review]
Looks really cool, excited for this 🚀  It seems like there is a tiny little bit of a problem with file reading part but other than that looks good to me!

File: forc-plugins/forc-client/src/op/submit.rs (hunk 3 above)
Comment (mitchmindtree) [reply forc-plugins/forc-client/src/op/submit.rs:33]: Ahh good catch!

In a future PR I'd like to add a section to our E2E testing that tests these CLI commands, but want to try making it a little easier to construct transaction files to test with first.

**Review by JoshuaBatty — DISMISSED** [review]

**Review by bingcicle — DISMISSED** [review]

**Review by kayagokalp — APPROVED** [review]

**Comment by mitchmindtree** [comment]
Okydoke just rebased and updated for the recent fuel-core-client changes, should be good to go again.

**Review by JoshuaBatty — APPROVED** [review]

**Review by kayagokalp — APPROVED** [review]

**Pull request merged**